contrib_bots/lib/gmail.py

```diff
@@ -0,0 +1,128 @@
+            '''
+
+    def triage_message(self, message, client):
+        # return True if we want to (possibly) response to this message
+
+        original_content = message['content']
+        self.bot_email = client.email
+
+        # This next line of code is defensive
+        if message['display_recipient'] == 'gmail':
+            return False
+        is_follow_up = original_content.startswith('@gmail')
+        return is_follow_up
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        values = parse_input(original_content)
+        body = values['body']
+        print('body: {}'.format(body)) 
+        label = values['label']
+        print('l: {}'.format(label)) 
+        subject = values['subject']
+        print('s: {}'.format(subject)) 
+        recipient = values['recipient']
+        print('r: {}'.format(recipient)) 
+
+        service = get_service()
+        print('got service')
+        email = create_message(body, subject, label, recipient)
+        result = send_message(service, email)
+
+        if (result == service.users().messages()):
+            client.send_message(dict(
+                type = 'private',
+                to = original_sender,
+                sender = self.bot_email,
+                content = 'Succesfully sent email to {}'.format(recipient)
+            ))
+        else:
+            client.send_message(dict(
+                type = 'private',
+                to = original_sender,
+                sender = self.bot_email,
+                content = 'Could not send email to {}'.format(recipient)
+            ))
+
+
+def parse_input(user_input):
+    body = user_input.split('--body')[1].strip()
+    content = user_input.split('--body')[0]  # added for readability
+    label = content.split('--label')[1].strip()
+    content = content.split('--label')[0]
+    subject = content.split('--subject')[1].strip()
+    content = content.split('--subject')[0]
+    recipient = content.split('--recipient')[1].strip()
+
+    return {'body': body, 'label': label, 'subject': subject, 'recipient': recipient}
+
+
+def create_message(body, subject, label, recipient):
+    message = MIMEText(body)
+    message['to'] = recipient
+    message['from'] = 'me'
+    message['subject'] = subject
+
+    return {'raw': base64.urlsafe_b64encode(message.as_string())}
+
+
+def send_message(service, email):
+    try:
+        result = service.users().messages().send(userID='me', body=email).execute()
+        return result
+    except errors.HttpError:
+        logging.exception('An error occured while sending the email.')
+        return -1
+
+
+def get_service():
+    try:
+        credential_path = os.path.join(os.path.expanduser('~'),
+                                       '.google-credentials.json')
+
+        store = Storage(credential_path)
+        credentials = store.get()
+
+    except IOError:
```

Comment:
This exception doesn't seem to be caught (because `store.get()` doesn't seem to be raising it).

Here's what happens for me if I haven't got any `.google-credentials.json`:

```
/srv/zulip-venv/local/lib/python2.7/site-packages/oauth2client/_helpers.py:255: UserWarning: Cannot access /home/zulipdev/.google-credentials.json: No such file or directory
  warnings.warn(_MISSING_FILE_MESSAGE.format(filename))
got credentials
Traceback (most recent call last):
  File "lib/gmail.py", line 132, in <module>
    get_service()
  File "lib/gmail.py", line 122, in get_service
    creds = credentials.authorize(httplib2.Http())
AttributeError: 'NoneType' object has no attribute 'authorize'

```